core/src/main/java/com/datastrato/gravitino/meta/UserEntity.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.meta;
+
+import com.datastrato.gravitino.Auditable;
+import com.datastrato.gravitino.Entity;
+import com.datastrato.gravitino.Field;
+import com.datastrato.gravitino.HasIdentifier;
+import com.datastrato.gravitino.Namespace;
+import com.google.common.collect.Maps;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import lombok.ToString;
+
+/** A class representing a user metadata entity in Gravitino. */
+@ToString
+public class UserEntity implements Entity, Auditable, HasIdentifier {
+
+  public static final Field ID =
+      Field.required("id", Long.class, " The unique id of the user entity.");
+  public static final Field NAME =
+      Field.required("name", String.class, "The name of the user entity.");
+  public static final Field PROPERTIES =
+      Field.optional("properties", Map.class, "The properties of the user entity.");
+  public static final Field AUDIT_INFO =
+      Field.required("audit_info", AuditInfo.class, "The audit details of the user entity.");
+
+  private Long id;
+  private String name;
+  private Namespace namespace;
+  private Map<String, String> properties;
+  private AuditInfo auditInfo;
+
+  private UserEntity() {}
+
+  /**
+   * Returns a map of fields and their corresponding values for this user entity.
+   *
+   * @return An unmodifiable map of the fields and values.
+   */
+  @Override
+  public Map<Field, Object> fields() {
+    Map<Field, Object> fields = Maps.newHashMap();
+    fields.put(ID, id);
+    fields.put(NAME, name);
+    fields.put(AUDIT_INFO, auditInfo);
+    fields.put(PROPERTIES, properties);
+
+    return Collections.unmodifiableMap(fields);
+  }
+
+  /**
+   * Returns the name of the user.
+   *
+   * @return The name of the user.
+   */
+  @Override
+  public String name() {
+    return name;
+  }
+
+  /**
+   * Returns the namespace of the user.
+   *
+   * @return The namespace of the user.
+   */
+  @Override
+  public Namespace namespace() {
+    return namespace;
+  }
```

Comment:
Why do we need namespace for `User`?

Reply:
User is under a metalake. So we should have a namespace.

Reply:
it's weird to use namespace here, we don't have a namespace concept for "User".

You can use metalake directly, or remove this interface. Since I don't see the usage of namespace from the current code.